### ISSUE TITLE
404 Not Found

### ISSUE THREAD

**Issue by FunkiestPeak** [NamelessMC/Nameless#152]
When I load up the NamelessMC 1.0.6, it asks if I am upgrading from 0.4.1, I click no because this is going to be a fresh install and it says, 404 Not Found The requested URL /install.php could not be found on this server. 

I don't know what to do this is my first time trying to install this.

**Comment by ThemInspectors** [comment]
> Error 1

Error: The requested URL was not found on this server

Meaning:

Your Apache webserver is not configured to allow .htaccess files

Fix:

Varies depending on the server configuration, however here is a guide for Ubuntu.

**Comment by Lmmb74** [comment]
I am having this error too, despite I have the module activated

**Comment by samerton** [comment]
If following all the steps in [this post](http://askubuntu.com/a/48363) don't help, there is a chance it may be installed in a subdirectory. 
You'll need to make sure it's in the either installed in the root directory of your webserver (usually /var/www or /var/www/html) or it's installed under a subdomain, however not a subdirectory.

**Comment by FunkiestPeak** [comment]
I all ready have set access to .htaccess files activated, and I have installed the namelessmc files into my root directory. I am using MAMP so my root is /Applications/MAMP/htdocs. I copied the files from the zip file straight into my htdocs or root folder.

**Comment by FunkiestPeak** [comment]
I also went looking around and found the install.php in the pages folder. I then copied install.php to htdocs and the next page loaded showing that I have all of the requirements. Then it moves to a blank configuration page.

**Comment by samerton** [comment]
The installer needs to be in the pages directory to function correctly, as it won't be able to find the required files in the htdocs folder.

You say you've enabled .htaccess files, were [these](http://stackoverflow.com/a/7670598) the steps you followed?

**Comment by FunkiestPeak** [comment]
Yes my file is like that except for one line at the bottom that I'm not sure if i need.

`<Directory "/Applications/MAMP/htdocs">
    #
    # Possible values for the Options directive are "None", "All",
    # or any combination of:
    #   Indexes Includes FollowSymLinks SymLinksifOwnerMatch ExecCGI MultiViews
    #
    # Note that "MultiViews" must be named *explicitly* --- "Options All"
    # doesn't give it to you.
    #
    # The Options directive is both complicated and important.  Please see
    # http://httpd.apache.org/docs/2.2/mod/core.html#options
    # for more information.
    #
		Options Indexes FollowSymLinks MultiViews
    Options All

    #
    # AllowOverride controls what directives may be placed in .htaccess files.
    # It can be "All", "None", or any combination of the keywords:
    #   Options FileInfo AuthConfig Limit
    #
    AllowOverride All

    #
    # Controls who can get stuff from this server.
    #
    Order allow,deny
    Allow from all

    XSendFilePath "/Applications/MAMP/htdocs"
</Directory>
`
Im not sure if the 'XSendFilePath "/Applications/MAMP/htdocs" ' is needed?

(Note: File was pre-made with the installation of MAMP, I did not add the Comments.)

**Comment by samerton** [comment]
I've not used MAMP before, I can try and install it at some point in the next few days to try and reproduce your issue.
If you're running it as a test environment on your local machine, I know that XAMPP has the correct setup by default, I'd recommend using that for now.

**Comment by FunkiestPeak** [comment]
Ok then thank you for your time.

**Comment by FunkiestPeak** [comment]
I will try to reinstall MAMP all together and if that doesn't work I will then install XAMPP.

**Comment by FunkiestPeak** [comment]
I tried to using XAMPP. It installed and when I go to run the servers everything starts except the apache server. Even though its says the apache server isn't up, when I click the button to go to the application click and ignore on the dialogue box saying not all servers are running, a page shows up saying "It works!". Would any of these errors have to do with me running these apps on a Mac OSX?

**Comment by FunkiestPeak** [comment]
Where is the code that tells the server to move from the Index in the root to install in pages?

**Comment by FunkiestPeak** [comment]
Can you send me a version of a working httpd.conf, and replace it with mine?

**Comment by samerton** [comment]
[This](http://stackoverflow.com/a/20351256) may help with your Apache issue with XAMPP. Once you've typed that command, try starting Apache through XAMPP again.

The code which tells the server where to find the install.php file is all within the .htaccess file and the index.php file (both in the root directory). The .htaccess file basically tells all requests coming in to visit the index.php file, which itself gets the correct file and displays it to the user. For this reason, the URLs are "friendly" ie they look like **your_url.com/forum** instead of **your_url.com/forum.php**

Unfortunately I can't send you a httpd.conf as I believe they vary quite a lot.

**Comment by FunkiestPeak** [comment]
I think I found the problem whenever I unzip the file there is no .htaccess files anywhere?

**Comment by FunkiestPeak** [comment]
YES!!!   This was the problem. ^^

The ,htaccess file was a hidden file. Meaning I had no way of seeing it or access it, or even copying it in the first place. I had to run this -
`defaults write com.apple.finder AppleShowAllFiles YES`
Within the Mac OSX Terminal. Then went back to the extracted folder and found the .htaccess file. I then copied it to my root. Thank you for your help Samerton, and very sorry for the trouble.
